extensions-core/kinesis-indexing-service/src/main/java/org/apache/druid/indexing/kinesis/KinesisRecordSupplier.java

```diff
@@ -637,6 +688,25 @@ public void close()
     this.closed = true;
   }
 
+  @VisibleForTesting
+  public Map<String, Long> getPartitionTimeLag()
+  {
+    return partitionResources.entrySet()
+                             .stream()
+                             .collect(
+                                 Collectors.toMap(k -> k.getKey().getPartitionId(), k -> k.getValue().getPartitionTimeLag())
+                             );
+  }
+
+  public Map<String, Long> getPartitionTimeLag(Map<String, String> currentOffsets)
+  {
+    return partitionResources.entrySet()
```

Comment:
The number of partitions can be on the order of hundreds, would it make sense to avoid streams here for performance (I remember various issues in the past around stream perf)